CONTRIBUTING.md

```diff
@@ -45,22 +45,132 @@ To add a new plugin you need to do the following.
+| **name**                         | The name of the Vim plugin.                                                                                                                                                                                  | Yes      | N/A              |
+| **url**                          | The URL of the plugin repository.                                                                                                                                                                            | Yes      | N/A              |
+| **colorscheme**                  | The name of the colorscheme.                                                                                                                                                                                 | No       | `name` parameter |
+| **deprecateExtraConfig**         | Flag to deprecate extra configuration.                                                                                                                                                                       | No       | `false`          |
+| **description**                  | A description of the plugin. Can also be used for non-normative documentation, warnings, tips and tricks.                                                                                                    | No       | `null`           |
+| **extraConfig**                  | Extra configuration for the plugin.                                                                                                                                                                          | No       | `cfg: {}`        |
+| **extraOptions**                 | Extra options for the plugin.                                                                                                                                                                                | No       | `{}`             |
+| **extraPackages**                | Extra packages to include.                                                                                                                                                                                   | No       | `[]`             |
+| **extraPlugins**                 | Extra plugins to include.                                                                                                                                                                                    | No       | `[]`             |
+| **globalPrefix**                 | Global prefix for the settings.                                                                                                                                                                              | No       | `""`             |
```

Comment:
I'll just copy past what you've written. I don't even know what a global prefix in `vim` is :joy: . 

Regarding the demonstration, this can be very useful, but I feel like it'll quickly overload the reader's ability to quickly do a search & find in this doc. He/She will be redirected to another nix example. Which is exactly what I don't want to do when coming for this section (for me at least).

Reply:
Maybe in a different doc or the global documentation in mdbook ?